pylsp_mypy/plugin.py

```diff
@@ -164,6 +166,23 @@ def pylsp_lint(
         return get_diagnostics(config, workspace, document, settings, is_saved)
 
 
+@lru_cache(maxsize=2000)
```

Comment:
this size is pretty arbitrary but also pretty safe in terms of memory usage. we definitely want to cache these results so we're not constantly calling out to another process.